internal/validate/vsa/storage.go

```diff
@@ -0,0 +1,186 @@
+// Copyright The Conforma Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package vsa
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// StorageBackend defines the interface for VSA storage implementations
+type StorageBackend interface {
+	Name() string
+	Upload(ctx context.Context, envelopeContent []byte, imageRef string) error
+}
+
+// SignerAwareUploader extends StorageBackend for backends that need access to the signer
+// (e.g., Rekor backend needs the public key for transparency log upload)
+type SignerAwareUploader interface {
+	StorageBackend
+	UploadWithSigner(ctx context.Context, envelopeContent []byte, imageRef string, signer *Signer) error
+}
+
+// StorageConfig represents parsed storage configuration
+type StorageConfig struct {
+	Backend    string            // rekor, local, etc.
```

Comment:
Since we don't have any others right now, maybe adjust the comment

```suggestion
	Backend    string            // rekor, local, (maybe other backend types in future?).
```